src/distance/ConfigurationRanker.cpp

```diff
@@ -0,0 +1,81 @@
+#include "aikido/distance/ConfigurationRanker.hpp"
+
+namespace {
+
+struct sortByCost
+{
+  bool operator()(
+      const std::pair<aikido::statespace::CartesianProduct::State*, double>&
+          left,
+      const std::pair<aikido::statespace::CartesianProduct::State*, double>&
+          right)
+  {
+    return left.second < right.second;
+  }
+};
+}
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+ConfigurationRanker::ConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton)
+  : mMetaSkeletonStateSpace(std::move(metaSkeletonStateSpace))
+  , mMetaSkeleton(std::move(metaSkeleton))
+{
+  if (!mMetaSkeletonStateSpace)
+    throw std::invalid_argument("MetaSkeletonStateSpace is nullptr.");
+
+  if (!mMetaSkeleton)
+    throw std::invalid_argument("MetaSkeleton is nullptr.");
+
+  mDistanceMetric = createDistanceMetricFor(
+      std::dynamic_pointer_cast<statespace::CartesianProduct>(
+          std::const_pointer_cast<statespace::dart::MetaSkeletonStateSpace>(
+              mMetaSkeletonStateSpace)));
+}
+
+//==============================================================================
+statespace::ConstStateSpacePtr ConfigurationRanker::getStateSpace() const
+{
+  return mMetaSkeletonStateSpace;
+}
+
+//==============================================================================
+std::vector<statespace::CartesianProduct::State*>
```

Comment:
It seems that we're returning raw pointers to the `CartesianProduct::State`s, rather than `CartesianProduct::ScopedState`s. What would happen if the input configurations are destructed before the ranking results are accessed?